static/app/components/core/button/index.tsx

```diff
@@ -24,7 +24,7 @@ type ButtonElement = HTMLButtonElement | HTMLAnchorElement;
 /**
  * Props shared across different types of button components
  */
-interface CommonButtonProps {
+export interface CommonButtonProps {
```

Comment:
shouldn't people be importing either `ButtonProps` or `LinkButtonProps`? I don't think we should export yet another thing.

This doesn't include the `to` field, so if people are using it they'll have to take special care with that field; which is what button/linkButtton is already doing right?

Reply:
There are some cases where the inner component is conditionally rendering a `Button` or `LinkButton` (Like the `UpgradeOrTrialButton` and `AddEventsCTA`), in which case we really shouldn't have an intersection of `LinkButtonProps | ButtonProps`